vscode/src/jdkDownloader.ts

```diff
@@ -29,6 +29,22 @@ import { promisify } from 'util';
 let customView: vscode.WebviewPanel;
 let logger: vscode.OutputChannel;
 
+export const getSha256Checksum = async (url: string) => {
+  return new Promise((resolve, reject) => {
+      https.get(`${url}.sha256`, res => {
+          let data = '';
```

Comment:
Perhaps you may add the check for the response status code 200 before attempting to read the data. Similar to the actual file download itself in the function `JDKDownloader`.